graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/db/migrations/V20230720161500_AddExtractorFragments.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.pipelineprocessor.rulebuilder.db.migrations;
+
+import com.google.common.collect.ImmutableList;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.rulebuilder.RuleBuilderFunctionGroup;
+import org.graylog.plugins.pipelineprocessor.rulebuilder.db.RuleFragment;
+import org.graylog.plugins.pipelineprocessor.rulebuilder.db.RuleFragmentService;
+import org.graylog2.migrations.Migration;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.bool;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.integer;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.string;
+
+public class V20230720161500_AddExtractorFragments extends Migration {
+
+    private static final Logger log = LoggerFactory.getLogger(V20230613154400_AddImplicitToStringFragments.class);
+    private final RuleFragmentService ruleFragmentService;
+    private final ClusterConfigService clusterConfigService;
+
+    @Inject
+    public V20230720161500_AddExtractorFragments(RuleFragmentService ruleFragmentService, ClusterConfigService clusterConfigService) {
+        this.ruleFragmentService = ruleFragmentService;
+        this.clusterConfigService = clusterConfigService;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2023-06-13T15:44:00Z");
+    }
+
+    @Override
+    public void upgrade() {
+        log.debug("Adding extractor fragments via migration");
+        if (Objects.nonNull(clusterConfigService.get(MigrationCompleted.class))) {
+            log.debug("Migration already completed!");
+//            return;
```

Comment:
After discussion with Maxwell, we decided to at least leave these commented out until next cycle for the 5.2 release because likely there might still be changes. I'll add a TO-DO to one of them